modules/jpa/src/main/java/io/oasp/module/jpa/common/base/AbstractGenericRevisionedDao.java

```diff
@@ -0,0 +1,95 @@
+/* Copyright (c) The m-m-m Team, Licensed under the Apache License, Version 2.0
+ * http://www.apache.org/licenses/LICENSE-2.0 */
+package io.oasp.module.jpa.common.base;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.sf.mmm.util.exception.api.ObjectNotFoundException;
+
+import org.hibernate.envers.AuditReader;
+import org.hibernate.envers.AuditReaderFactory;
+
+import io.oasp.module.jpa.dataaccess.api.GenericRevisionedDao;
+import io.oasp.module.jpa.dataaccess.api.MutablePersistenceEntity;
+import io.oasp.module.jpa.dataaccess.api.RevisionMetadata;
+import io.oasp.module.jpa.dataaccess.impl.LazyRevisionMetadata;
+
+/**
+ * This is the abstract base-implementation of a {@link AbstractGenericDao} using to manage the revision-control.
+ *
+ * @param <ID> is the type of the {@link MutablePersistenceEntity#getId() primary key} of the managed
+ *        {@link MutablePersistenceEntity entity}.
+ * @param <ENTITY> is the {@link #getEntityClass() type} of the managed entity.
+ *
+ * @deprecated It is only provided to support old functionality of DAO, instead use
+ *             io.oasp.module.jpa.dataaccess.base.AbstractGenericRevisionedDao instead
+ */
+@Deprecated
+public abstract class AbstractGenericRevisionedDao<ID, ENTITY extends MutablePersistenceEntity<ID>>
```

Comment:
This class was IMHO before present in `oasp4j-jpa-envers` and I would still expect it there. This is completely envers specific.

Reply:
Basically what i found below hierarchy here in mtsj that
 BookingDaoImpl->ApplicationDaoImpl->AbstractRevisionedDao->AbstractGenericRevisionedDao->AbstractGenericDao

So I tried to replicate this one, if i tried to use oasp4j-jpa-envers than it would call AbstractGenericDao from oasp4j-jpa-dao 

So i prepared BookingDaoImpl->LegacyApplicationDaoImpl-->AbstractRevisionedDao->AbstractGenericRevisionedDao->AbstractGenericDao (all from oasp4j-jpa)

I thought of directly connecting from from LegacyApplicationDaoImpl to AbstractGenericDao  but not aware of the implication of skipping  oasp4j-jpa-envers

So basically now BookingDaoImpl extends LegacyApplicationDaoImpl and can use findPaginated method as it is and would required minimal changes in mtsj